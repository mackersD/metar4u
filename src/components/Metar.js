import React from 'react'
import Vector from './Vector'
import MetarText from './MetarText'

const Metar = ({distance, direction, report, isFetching, error}) => {
  const styles = {
    display: "flex",
    "flexDirection": "row",
    "alignItems": "center"
  }
  return (
    <div style={styles}>
      <Vector
        distance={distance}
        direction={direction}
      />
      <MetarText
        text={report}
      />
    </div>
  )
}

export default Metar
