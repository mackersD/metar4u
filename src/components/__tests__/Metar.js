import React from 'react'
import Provider from 'react-redux'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Metar from '../Metar'

const fakeStore = (state) => {
  return {
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => { return Object.assign({},state) }
  }
}

describe('Component - Metar', () => {
  it('renders', () => {
    const comp = renderer.create(
      <Metar store={fakeStore({})} />
    ).toJSON()

    expect(comp).toMatchSnapshot()
  })
})
