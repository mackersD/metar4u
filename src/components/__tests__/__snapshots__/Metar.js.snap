// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Metar renders 1`] = `
<div
  className="metar"
>
  <div
    className="spinner"
  >
    <div
      className="vector"
    >
      <div
        className="vectorArrow"
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
            "display": "inline-block",
            "msTransform": "rotate(0deg)",
            "transform": "rotate(0deg)",
          }
        }
      >
        ^
      </div>
      <div
        className="vectorDistance"
      >
        0
      </div>
    </div>
    <div
      className="metarText"
    />
  </div>
</div>
`;
